### PR TITLE
Commit statuses: allow being explicit

### LINKED ISSUE #167: Commit Statuses: Allow configuring which statuses are expected / required
[`ci-status--required`](https://github.com/iFixit/pulldasher/compare/ci-status--required)

We'd like to be able to explicitly list which commit status are required before we mark a pull as green. This list should be per-repo.

### PR DESCRIPTION
**Pull request by danielbeardsley**
Allow configuring the required commit statuses per repo so we don't
accidentally show something as green when it's missing some CI statuses.

This is backward compatible, existing configs will be interpreted in the
same way. Implemented this by sending the list of repos with their
requirements on the initial socker.io connection.

The config docs say it well:

    // Just listing the repo as a string indicates pulldasher should consider
    // CI passing if all the statuses that exist are passing and that there
    // is at least one.
    "owner/repo",
    // Listing them like this allows being more explicit with the commit
    // statuses
    {name: "owner/otherRepo", requiredStatuses: ["tests", "build", "codeClimate"]}

Closes #167

### CODE REVIEW

File: lib/authentication.js [hunk 1]
```diff
@@ -76,6 +80,10 @@ module.exports = {
       // Ensure index is authenticated.
       app.get('/', function (req, res, next) {
          if (req.isAuthenticated()) { return next(); }
+         if (process.env.MOCK_AUTH_AS_USER) {
+            req.user = {username: process.env.MOCK_AUTH_AS_USER};
```
Comment (sterlinghirsh) [inline lib/authentication.js:84]:
```suggestion
         if (FAKE_USER) {
            req.user = {username: FAKE_USER};
```

File: lib/config.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+var configPath = process.env.CONFIG_PATH || '../config';
```
Comment (sterlinghirsh) [inline lib/config.js:1]: It's a little confusing to have 2 config.js files. lib/config.js requires ../config.js. Anyone loading the config file directly may accidentally miss this file, and it's a hard mistake to spot.

**Comment by sterlinghirsh** [comment]
CR :zap: dev_block :zap: on a few small comments. Feel free to not rename config.js if you don't think it's worth.

File: lib/config.js (hunk 2 above)
Comment (danielbeardsley) [reply lib/config.js:1]: Good point. I'll rename this one.

**Comment by danielbeardsley** [comment]
QA 👍 

Ran this in development and tested several different iterations of `requiredStatuses` and they all worked.

**Comment by danielbeardsley** [comment]
un_dev_block 👍

File: lib/authentication.js (hunk 1 above)
Comment (danielbeardsley) [reply lib/authentication.js]: Fixed!

**Comment by sterlinghirsh** [comment]
CR :zap:

**Comment by danielbeardsley** [comment]
Re QA 👍

**Comment by cdcline** [comment]
CR 👍

**Comment by jarstelfox** [comment]
Once this is deployed what is the process to upgrade pulldasher?

**Comment by addison-grant** [comment]
> Once this is deployed what is the process to upgrade pulldasher?

Steps 3 to 5 in https://overlord.dozuki.com/c/Pulldasher#Section_Adding_a_repo

**Pull request merged**